interop/klr/fromNKI.hpp

```diff
@@ -0,0 +1,511 @@
+#pragma once
+
+#include "klir_ast.hpp"
+#include "klir_common.hpp"
+#include <variant>
+#include <string>
+#include <vector>
+#include <optional>
+
+namespace Trace {
+
+struct Error {
+    std::string msg;
+    Error(std::string&& m) : msg(std::move(m)) {}
```

Comment:
If you just take the string by value, it's way easier for users (and just as efficient). They can call it like: `Error("oh no")` instead of `Error(std::move(std::string("oh no")))`. I see you discovered that and made a `TraceHelper(std::string msg)` helper function later to get the same result as what I'm suggesting here. You wouldn't need that helper anymore

The only advantage of the "r-value only" version you made here is that users are forced to call move() themselves and can't accidentally do a copy instead. But you can see the downside, as you had to make a helper function to work around the clunkiness.

```suggestion
    Error(std::string m) : msg(std::move(m)) {}
```